### PR TITLE
Extensionality of globular types

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Adds a postulate to characterize equality of globular types.

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
It is not presently clear to me whether it is possible to show that equality of globular types as introduced in this PR is equivalent to coinductive equalities of globular types, and therefore equivalences of globular types by univalence.

**Comment by fredrik-bakke** [comment]
The idea of this PR arose from reading the introduction to this paper:

- _Formalising inductive and coinductive containers_, by Stefanie Damato, Thorsten Altenkirch, and Axel Ljungström ([arXiv:2409.02603](https://arxiv.org/abs/2409.02603))

**Comment by fredrik-bakke** [comment]
Quite a few questions linger with this PR. First and foremost, is the added axiom validated in all ∞-topoi?
I'm currently working on an accompanying contribution showing that globular types and globular diagrams are equivalent, using this axiom.

File: src/structured-types/equality-globular-types.lagda.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+```agda
+record
+  Eq-Globular-Type
+  {l1 l2 : Level} (A : Globular-Type l1 l2) (B : Globular-Type l1 l2)
+  : UU (lsuc l1 ⊔ lsuc l2)
+  where
+  coinductive
+  field
+    0-cell-Eq-Globular-Type :
+      0-cell-Globular-Type A ＝ 0-cell-Globular-Type B
```
Comment (EgbertRijke) [inline src/structured-types/equality-globular-types.lagda.md:53]: Shouldn't extensionality of globular types have equivalences here?

File: src/structured-types/equality-globular-types.lagda.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+## Definitions
+
+### Coinductive equality of globular types
+
+```agda
+record
+  Eq-Globular-Type
+  {l1 l2 : Level} (A : Globular-Type l1 l2) (B : Globular-Type l1 l2)
+  : UU (lsuc l1 ⊔ lsuc l2)
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/structured-types/equality-globular-types.lagda.md:48]: oops, that was probably a typo

File: src/structured-types/equality-globular-types.lagda.md (hunk 1 above)
Comment (fredrik-bakke) [reply src/structured-types/equality-globular-types.lagda.md:53]: The point of phrasing it this way is to make the postulate be independent of univalence

File: src/structured-types/equality-globular-types.lagda.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+</details>
+
+## Idea
+
+We postulate that [equality](foundation-core.identity-types.md) of
+[globular types](structured-types.globular-types.md) is characterized by
+equality of the 0-cells together with, coinductively, a binary family of
+equalities of the globular type of 1-cells over the equality of the 0-cells.
+This phrasing is used so that the extensionality principle is independent of
+[univalence](foundation.univalence.md).
```
Comment (EgbertRijke) [inline src/structured-types/equality-globular-types.lagda.md:38]: I would expect extensionality of globular types to say that the canonical map from equality to globular equivalences is an equivalence

File: src/structured-types/equality-globular-types.lagda.md (hunk 1 above)
Comment (EgbertRijke) [reply src/structured-types/equality-globular-types.lagda.md:53]: I see. Ok, your point is well taken.

**Comment by EgbertRijke** [comment]
I don't think the open question is really open at all. The equality relation you have given is just bisimilation, and for coinductive types in a type theory with function extensionality it is known that bisimilation characterizes the identity type. Basically all you need to know is that the coinductive type is the limit of iterated applications of a polynomial endofunctor to the terminal object, and that identity types of limits are limits of identity types.

**Comment by EgbertRijke** [comment]
I'd be happy to merge this PR once it passes the checks

**Comment by fredrik-bakke** [comment]
> I don't think the open question is really open at all. The equality relation you have given is just bisimilation, and for coinductive types in a type theory with function extensionality it is known that bisimilation characterizes the identity type. Basically all you need to know is that the coinductive type is the limit of iterated applications of a polynomial endofunctor to the terminal object, and that identity types of limits are limits of identity types.

Oh, well that settles it then :) Thank you for the answer

**Pull request merged**